### ISSUE TITLE
`invalid-attribute-override` - false negative when override changes the type

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3634]
### Summary

`main.py`:

```py
from typing import override


class Foo:
    foo: int = 1

class Bar(Foo):
    foo: str = "" # no error
```

[Playground](https://play.ty.dev/3771f3e9-111e-4b6a-9614-f7a03ff6888c)

[the docs](https://docs.astral.sh/ty/reference/rules/#invalid-attribute-override) mention this limitation:

> This rule currently only covers class-variable and instance-variable category changes.

but i couldn't find an issue for this

### Version

1a34347c2
